### ISSUE TITLE
Can't open .cshtml files with the Visual Studio 2015 HTML Editor (post ASP.NET 5 RC1 Update 1)

### ISSUE THREAD

**Issue by Kaelum** [aspnet/Tooling#390]
On a brand new machine with Visual Studio 2015 Enterprise with Update 1 and [Microsoft ASP.NET and Web Tools](https://visualstudiogallery.msdn.microsoft.com/c94a02e9-f2e9-4bad-a952-a63a967e3935) (ASP.NET 5 RC1 Update 1) installed, I can't open any .cshtml file with the HTML Editor.  I had this working on other machines a few weeks back, but now none of them work either, and **none of the steps listed under issue #276 work**.

Removing the update allows the pages to load, but then I get red squiggles in the .cshtml page that refer to missing DLLs.  Everything builds, and the application runs under IIS, but the editor is FUBAR.

[sayedihashimi](https://github.com/sayedihashimi) asked if anyone is still having issues, but the thread is now closed, so I opened this one.

**Comment by ThomasSchmidt** [comment]
Just experienced the same issue after updating with Visual Studio Professional 2015 Update 1.Tried all workarounds in #276 but nothing is working. Have tried to disable all extensions, like ReSharper and so on, but still no luck getting VS to open razor views.

**Comment by Kaelum** [comment]
I opened a ticket on Microsoft Connect for this as well, as this is affecting several products, including Azure:

https://connect.microsoft.com/VisualStudio/feedback/details/2323617/product-stability-asp-net-5-rc1-update-1

**Comment by PilotC150** [comment]
I'm also having this issue after installing Update 1 RC 1.  Changing the default editor for .cshtml files works to open and edit them, but the razor syntax isn't correctly highlighted.

**Comment by Leglaw** [comment]
I can't get this resolved following any of the suggestions in this thread or in #276. Aside from completely uninstalling and reinstalling 2015, not sure what else to try.

**Comment by mittork** [comment]
@Leglaw same here

**Comment by jdhorner** [comment]
I have the _**exact**_ same problem as @Kaelum. I cannot find a way to revert/downgrade/uninstall ASP.NET 5 RC1 Update 1

**Comment by sayedihashimi** [comment]
Assigned to @madskristensen

**Comment by Thieum** [comment]
:+1: 
The workaround posted in the other thread fixed the problem so far:

1. Close VS 
2. Delete the content of %LocalAppData%\Microsoft\VisualStudio\14.0\ComponentModelCache
3. Open VS

**Comment by Kaelum** [comment]
@Thieum: If you had bothered to read my post, and the post of others, you would have noticed that that does not work for anyone in this thread.  In fact, NONE of the work-arounds/solutions posted ANYWHERE work.  AND to top it off, Microsoft is pushing this out to the general public, as it is hidden inside of other updates.

**Comment by jdhorner** [comment]
## **NONE** of these worked: 

- Removing ComponentModelCache
- Using `/ResetUserData`
- Using `/ResetSettings`
- Starting in safe mode and trying to open a `.cshtml` file
- Any of the `dvnm` commands listed in other threads
- Repairing the VS installation
- Repairing the ASP.NET 5 RC1 Update1 installation

I had to completely uninstall every single item listed in the Remove Programs window that had the words "Azure", ".NET", or "Visual Studio". 

Even then, the `ASP.NET 5 RC1 Update1` item in the list (which was listed twice, by the way, with slightly different internal version numbers) didn't uninstall. The lower-versioned one claimed to successfully remove itself, but it never removed from the list, and never actually uninstalled. 

The higher-versioned one gave an error upon clicking uninstall, and then prompted a file explorer window asking for the update installation .exe. ... So I had to go find and download it from [asp.net](https://get.asp.net/), point the uninstaller to it, and then it finally removed itself. 

Only then could I completely reinstall Visual Studio 2015 community edition, load my solution, and actually double-click-to-view `.cshtml` files. Lo and behold, a warning to you all: the [same three updates](https://imgur.com/fkc4vNY) are back. ONE of them is the disaster-causing culprit.

**Comment by mittork** [comment]
So, is there any estimation when this will be fixed? In my team now almost all people have this issue.... Using the HTML WebForms editor is possible, but not really useful. BUT a compare/merge with TFS is NOT possible and that is almost a critical thing.....

**Comment by Leglaw** [comment]
@jdhorner That is precisely what happened to me -- with particular interest in the double install and having to point the installer to the exe so it could execute the uninstall before being able to reinstall VS2015.

**Comment by Eilon** [comment]
@barrytang bump on this thread (see https://github.com/aspnet/live.asp.net/issues/80 as well).

**Comment by Kaelum** [comment]
FYI, this can be replicated with 100% success using the following steps.

1. Start with a clean install of Windows 10.
2. Install Visual Studio Enterprise 2015 with Update 1 from the ISO, with Select All.
3. Open Visual Studio 2015.
4. Open an ASP.NET MVC application
5. Open any .cshtml file. (this works)
6. Close your solution.
7. Navigate to TOOLS / Extensions and Updates... / Updates.
8. Update everything, until there is nothing left to update.
9. Restart Visual Studio 2015.
10. Open the same ASP.NET MVC application that you had opened before.
11. Open any .cshtml file.
12. <errors> 2 different popups.
13. Open the Control Panel.
14. Open Programs and Features.
15. You may see 2 installations of "Microsoft ASP.NET 5 RC1 Update 1", build 11125 (from Microsoft Azure SDK for .NET (VS2015)), and build 11123 ([Microsoft ASP.NET and Web Tools](https://visualstudiogallery.msdn.microsoft.com/c94a02e9-f2e9-4bad-a952-a63a967e3935)).  This may depend upon the order that extensions are updated, but doesn't affect the end result.

**Comment by Leglaw** [comment]
@Kaelum Wow that's extremely valuable. I've not tried to reproduce yet myself, but makes me wonder, does MS pay bounty on bug finding / squashing?

I became so extremely frustrated with the whole endeavor having wasted so much time on it that I just completely uninstalled my VS2015 and reinstalled, cutting my losses short I figured.

Anyway, I wanted to say thanks in case no one else did.

**Comment by Eilon** [comment]
@Kaelum an update: we think it's probably a setup issue and we're having our set experts take a look at this ASAP.

**Comment by mlorbetske** [comment]
@Kaelum thank you for the stack traces and the list of things you've tried. I know you've attempted opening the file in safe mode (which should disable any 3rd party extensions), but do you have any 3rd party extensions installed? If so, can you try disabling all extensions and opening a cshtml file? I'm following your install steps to hopefully get a machine into the state where the issue reproduces so we can debug it directly.

**Comment by jdhorner** [comment]
@mlorbetske for what it's worth, I had the same setup as described as @Kaelum, and even in safe mode, .cshtml files would not open.

**Comment by mlorbetske** [comment]
@jdhorner thanks, and thanks for the breakdown of what you've tried as well.

**Comment by Kaelum** [comment]
@mlorbetske: Just like @jdhorner, I had tried everything as well.  The only difference that I have had is that NOTHING fixes the issue, short of wyping out Windows and starting from scratch and making sure that nothing installs **ASP.NET 5 RC1 Update 1**.  All of my machines are brand new installations of Windows, with the exception of one Windows 7 laptop, which also exhibits the exact same behavior.  On all of them, I started with the VS 2015 Update 1 media, not the RTM media.  From an observation standpoint, it appears that if you started with the RTM version of VS 2015 and upgraded from there, you can get your installation to work.  If you started with the Update 1 media, none of the work-arounds work.

**Comment by antonioortizpola** [comment]
Same problem here, nothing worked, luckily i have a 2013 licence, but it is really frustrating the need to  open other VS with the same project just to edit Cshtml. When i try to uninstall ASP.NET 5 it shows "Setup locked" and shows a dialog to open a .exe file, if i click repair then the setup gets stuck

**Comment by mlorbetske** [comment]
Sorry for the delayed reply. I've been trying permutations of the update installers using Win10 + VS2015.1 (ISO from MSDN, not VS2015 RTM + Update 1) as a starting point but, unfortunately, have yet to be able to reproduce the issue. @Kaelum @jdhorner @antonioortizpola could one or more of you run the setup log collection tool from [here](https://www.microsoft.com/en-us/download/details.aspx?id=12493) and email me at mlorbe at microsoft dot com so we can figure out how to get them?

**Comment by Kaelum** [comment]
@mlorbetske is there any update on this?  In about 1.5 weeks, I need to work on a prototype for a project that will take 3 years to complete, and using 4.6.x is not an option.  Thus far, I have not been able to build any machine from a clean install of Windows (or otherwise) that does not run into this issue, so I don't know how you are able to do so.

**Comment by mlorbetske** [comment]
@Kaelum unfortunately I don't have an update yet, the person who investigates things on the setup side is currently tied up with preparing things for //build/. I have tried quite a few different permutations of update install orders (using fresh Win10 VMs and the VS 2015+U1 ISO from MSDN as the starting point) and have yet to be able to reproduce the issue. I wish I could so we could get to the bottom of this, but no success yet.

Exact steps followed:
* Setup a new VM with Windows 10 RTM and nothing else
* Apply all Windows updates
* Install VS2015+U1 from the MSDN ISO, selecting all options during setup
* Create a new ASP.NET MVC project (non-DNX), verify that cshtml files can be opened
* Apply all updates from Extensions and Updates, restarting as necessary
* Open VS and open the solution containing the MVC project again
* Open a cshtml file

Is there a small repro project that I could get? Perhaps it has something to do with the particular version of MVC being used in the project.

**Comment by Kaelum** [comment]
@mlorbetske I have some better news.  Today, I installed the third version of [AspNet5.ENU.RC1_Update1.exe](https://visualstudiogallery.msdn.microsoft.com/c94a02e9-f2e9-4bad-a952-a63a967e3935) (1.0.20204.0) on my home workstation, and as all of the other installations it was still broken.  It also still creates 2 entries in Programs and Features:

![microsoft asp net 5 rc1 update 1](https://cloud.githubusercontent.com/assets/1742782/13905785/de2eac76-ee86-11e5-81b2-8dedeca72b0b.png)

However, this time I tried using /safemode again, and IT WORKED!  I'm going to try and find out which add-on is causing the problem, if that is the issue.  If it is an add-on, it's one of the @madskristensen add-ons, as they are the only 3rd party add-ons that I use, and they come with VS 2015 now.

**Comment by Kaelum** [comment]
@mlorbetske I ran additional tests, and it only works in /safemode.  If you switch out of /safemode it will never work, and it will break all future attempts to use /safemode until you delete the ~/ComponentModelCache folder.  All of these tests were made w/o any 3rd part extensions installed.  Also, when in /safemode, an error popup window does display, but it is able to open the .cshtml:

```
System.InvalidOperationException: value at Roslyn.Utilities.Contract.ThrowIfNull[T](T value, String message)
  at Microsoft.VisualStudio.LanguageServices.Implementation.Interop.ComAggregate.GetManagedObject[T](Object value)
  at Microsoft.VisualStudio.LanguageServices.TypeScript.VsPackage.WpfTextViewConnectionListener.SubjectBuffersConnected(IWpfTextView textView, ConnectionReason reason, Collection`1 subjectBuffers)
  at Microsoft.VisualStudio.Text.Editor.Implementation.ConnectionManager.<>c__DisplayClass7_0.<OnGraphBuffersChanged>b__0()
  at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(Object errorSource, Action call)
```

There are several:

> Exception  parsing &lt;name of module&gt; (under KEY_LOCAL_MACHINE\Software\Microsoft\.NETFramework) as a version

entries as well.

**Comment by mlorbetske** [comment]
@Kaelum glad to hear that the situation has improved a bit, could you send the activity log (from %appdata%\Microsoft\VisualStudio\14.0\activitylog.xml) please? If safemode is helping then some things must be being skipped during the safemode, the activity log will help us figure out what those things are.

**Comment by KimHolzmann** [comment]
If I have the files included in the project it works.
But I if will open a file that's not included vs2015 crashes.

Mabye it can help others a litte in running their coding
AND MS to fix the problem.
And it's not only cshtml files!

Best regards 
Kim Holzmann

![image](https://cloud.githubusercontent.com/assets/8200579/13914635/e73935ce-ef4f-11e5-8ee0-0977dcdeeb8e.png)
![image](https://cloud.githubusercontent.com/assets/8200579/13914652/000703c4-ef50-11e5-9203-b3a8088e6881.png)

**Comment by Kaelum** [comment]
Hey @mlorbetske, I'm using my work machine now, which does have add-ons, but it is exhibiting the same behavior.  Here are the activity logs for this machine.  I can get them for the other machine later this evening, if you need them.

[ActivityLogs.zip](https://github.com/aspnet/Tooling/files/182883/ActivityLogs.zip)

**Comment by Kaelum** [comment]
@KimHolzmann this thread is about a completely different issue than the one you are experiencing.  There is another issue opened for that issue, but I don't know which one it is.  Please move your post to that issue.

**Comment by shamglam** [comment]
Just wanted to add that I'm experiencing the same thing as @Kaelum. Deleting the cache and starting in safe mode is the only thing which seems to make cshtml files work.

**Comment by mlorbetske** [comment]
@Kaelum thanks, we started going over the logs last night and we've got some ideas - too early to say for certain if it is what we think it might be yet though.

**Comment by KimHolzmann** [comment]
@Kaelum Fair enough - I didn't  see the different so clearly, still don't sorry about that, the post will be removed.

**Comment by Kaelum** [comment]
@KimHolzmann np, the issue you want is 389, it's closed, as I think it's been fixed.

**Comment by GearTheWorld** [comment]
Any news on a fix for updating GitHub which results in ComponentModelCache folder to be garbage?
Have to go to %LcoalAppData%\Microsoft\VisualStudio\14.0 and delete this folder each time.  For your information once Visual Studio Restarts, there's a dialog saying there's a problem with the configuration setting change, Continue to show this message?

**Comment by Kaelum** [comment]
@GearTheWorld the fix for this thread may, or may not, fix your issue.  On this thread, VS2015 doesn't work at all with .cshtml files, unless you are in safe mode.  If you are working with VS in a non-safe mode, you may have an unrelated issue.  We are not getting configuration errors, which may mean that your .settings file is corrupted.

**Comment by GearTheWorld** [comment]
In that case my problem is indeed not related.  Thank you!

**Comment by AlexEyler** [comment]
@GearTheWorld what issue are you referring to? We've made some improvements to MEF in regards to caching in Update 2 - I hope those will help unblock you. Otherwise, feel free to reach out to me directly.

**Comment by GearTheWorld** [comment]
@AlexEyler The problem I had is simply the same each time I update GitHub.  So recently I updated it via the Notification in Visual Studio 2015 Enterprise Update 1 and when it asks to restart VS, I do and I'm back with a dialog saying there was a **configuration setting problem **sorry for not being precise on this one then I have the choice to stop showing this message which I select YES then once my MVC 5 project is fully loaded I'm unable to open a .cshtml because I receive either "The operation could not be completed" or "Object reference not set to an instance of an object" so the only alternative I know and I'm happy to know it because the first time I had this issue I had to search the web a lot to finally find a Stack Overflow solution that told me to delete the famous ComponentModelCache folder to solve it.

**Comment by Kaelum** [comment]
@mlorbetske I installed Update 2 on one of my workstations that had VS2015 previously installed, but it didn't change anything.  I have a new workstation that doesn't have VS installed, but I won't be able to work with it until this weekend.  I'll let  you know those results once I know.  You're probably at Build, but is there any progress with the logs?  I've been watching the live stream from Build, and really want to start working with tis stuff.  :)

**Comment by Kaelum** [comment]
@GearTheWorld BTW, did you look at #276?  They closed the thread, but it was more about the issues that you are describing.

**Comment by mlorbetske** [comment]
@Kaelum unfortunately nothing definitive yet as to the root cause, but we have used the information you've given us in to put fixes in RC2 that should both unblock this scenario and get us more information about what was happening to cause the problem in the first place.

**Comment by tobiasbunyan** [comment]
Spent a while pulling my hair out over this the last couple of days and finally solved it.  It's nothing related to the VS environment directly in my case but it might be worth mentioning in case some other people have this problem.

I noticed that, for me, I could create new mvc projects ok and indeed copy the files which wouldn't open in one project into another and could open them from there.

It turns out that the problem is the setting for the web pages version in the web.config file.  Evidently this must be read when you open a cshtml file so that vs knows what sort of page it is parsing.  Mine was set to 5:

```
   <appSettings>
    <add key="webpages:Version" value="5.0.0.0" />

```
When in all other projects it was still 3

```
  <appSettings>
    <add key="webpages:Version" value="3.0.0.0" />

```
Changing the 5 to 3 enabled opening of the files again.

So basically make sure it isn't a config error preventing opening of the files.

Some background reading:


How does VS determine razor versions:
https://blogs.msdn.microsoft.com/webdev/2013/07/10/how-does-vs-determine-which-version-of-razor-engine-to-use-when-editing-razor-webpage-files/

**Comment by Eilon** [comment]
@tobiasbunyan that's for ASP.NET 4.x, but this discussion here is about ASP.NET Core (formerly ASP.NET 5).

**Comment by mandazi** [comment]
FWIW, I recently updated Visual Studio to VS 2015 Version 2 and ASP.NET Web Tools Update 1 RC1 and got the same error when trying to load any .cshtml file in VS.  What fixed it was updating the Azure SDK.

**Comment by Kaelum** [comment]
@mlorbetske what is the current status?  Our CTO is extremely concerned that the RTM release of VS 2015 has "hidden" beta code that comes along with it, and that none of our web application engineers are able to use VS 2015 as a result of this.  I can't use his words, but wants to know who approved the release of beta code within RTM products, that is not a completely separate install?

None of us knowingly installed RC1, as it is hidden inside the RTM releases of several other products, mostly the Azure products.  We had been considering Azure, but it is no longer a 5 letter word around here.  It has become a very specific 4 letter word.

The bottom line is that we need a fix immediately, not 4-6 weeks down the road, which may not even address our issue.  How should I handle this going further?  Can we get a Hotfix here?  Or do we need to go through other channels?  I have an open Connect case, but I was redirected to here.

**Comment by Kaelum** [comment]
@mlorbetske: P.S. He also wants to know what assurances do we have that beta code will never again be hidden in RTM releases.

**Comment by mlorbetske** [comment]
@Kaelum sorry for the lack of update. We're still where we were, making sure that we don't block opening the file in the cases that the traces and dumps have indicated the failure and diagnostics being added for those code paths so we can figure out what's happening if it still happens. Unfortunately, none of the configurations we've tried have experienced this issue still - we're able to repro MEF cache corruption issues and the "suggested extensions" issue, but, unfortunately, still have not been able to repro this one.

With regard to the questions and commentary around the inclusion of beta software in an RTM product, the possibility of early access to the bits (described above) that should suppress/help us more fully diagnose the original cause of the issue and the loss of consideration for using Azure services, I've brought this thread to the attention of my manager - we'll be discussing each of these things today.

**Comment by barrytang** [comment]
Hi @Kaelum, 

I know @mlorbetske is already working with you offline but here is what I tried.

I tried the steps your shared previously with VS2015 Update 2 ISO and I wasn't able to repro (it looks like VS2015 Update 1 ISO no longer works and that's why I moved to Update 2 ISO).

![image](https://cloud.githubusercontent.com/assets/1648865/14804474/a02b00c4-0b18-11e6-8694-333ad687d12d.png)

I know you said you installed Update 2 on top of Update 1 but I wonder if installing Update 2 on top of the Update 1 ISO somehow put your machine in a bad state. Can you try your steps again with the Update 2 ISO from the following and see if you can still repro?

https://www.visualstudio.com/downloads/download-visual-studio-vs

![image](https://cloud.githubusercontent.com/assets/1648865/14804697/74089716-0b1a-11e6-8b52-d7bab5eb4e66.png)

Can you also let me know if you followed the "How to install files offline" instructions? I didn't do it when I tried to repro.

https://msdn.microsoft.com/en-us/library/e2h7fzkw(v=vs.140).aspx#bkmk_offline

If you can still repro the issue, can you download the Collect.exe tools from http://aka.ms/vscollect, run it from a command line and share the vslogs.zip with us?

Thanks and sorry for the trouble,
Barry

**Comment by Kaelum** [comment]
@barrytang I think you may have misunderstood.  I have an Enterprise MSDN account, from which I downloaded the ISO, as had everyone else in our enterprise.  I have a new machine that I will be trying the latest installation (Update 2?) on once I get a chance.  As for installing Update 2 over Update 1, it was the Update 1 installation that became corrupted once the Azure SDK updates installed, and installing Update 2 allowed the opening of .cshtml files while in /safemode only, which is useless.

Aside from that, there is only one thing that I am aware of that would be different from your installations, and those of ours.  We aren't doing this in a VM, we are installing directly onto the host machine.  This shouldn't make a difference, but I have seen some things react differently when installed onto a VM as opposed to a physical machine.

I've been working with @mlorbetske directly, and I'll let you both know how the install on this new machine goes.  One thing that I plan to do is remove everything Azure once the initial installation is complete, and then run the updates, excluding RC1.  If Update 2 has RC1 in it, please let me know now, so that I can use Update 1 instead.

**Comment by barrytang** [comment]
>One thing that I plan to do is remove everything Azure once the initial installation is complete, and then run the updates, excluding RC1. 
You might want to skip the Azure SDK update instead if you are not using the Azure related tooling features.

>If Update 2 has RC1 in it, please let me know now, so that I can use Update 1 instead.
No, Update 2 has the same Web Tools bits as Update 1.

**Comment by Kaelum** [comment]
@mlorbetske & @barrytang on a fully patched, clean install of Windows 10 Enterprise, I installed Office 2016 Professional, SQL Server 2014 Developer, made sure that everything was working perfectly for a few days, and then I installed Visual Studio 2015 Enterprise with Update 2.  Immediately after the installation, without any additional patches, I started VS 2015, setup my connections to the TFS servers that I use, and I loaded the MVC 5 solution that I've been working on for the past 2 years.

The .cshtml files open, but Intellisense is totally FUBAR, even after building the solution.  This is exactly the state that I end up in after uninstalling "ASP.NET 5 RC1 Update 1", which in this case was never installed.  So, I decided to install the patch, Azure SDK 2015 2.9, which has "ASP.NET 5 RC1 Update 1" hidden inside of it.  As expected, immediately after installing it, .cshtml files can no longer be opened.  However, I now know that it is the "preview" feature that is crashing with the first error, and that attempting to open the file for editting returns a NullReferenceException.

Next, I removed everything Azure, as we don't need, nor do we want any of it installed, which finally removes all the notifications to install Azure updates.  But it doesn't fix the problem.  So, the bottom line is that a Complete install of Visual Studio 2015 with Update 1 appears to work out of the box, Update 2 does not.  And after installing "ASP.NET 5 RC1 Update 1", neither installation will work from that point on even if it is removed.

**Comment by barrytang** [comment]
> However, I now know that it is the "preview" feature that is crashing with the first error, and that attempting to open the file for editting returns a NullReferenceException.

Can you see if this is the same issue as [this](https://blogs.msdn.microsoft.com/webdev/2016/01/12/visual-studio-keeps-showing-suggested-extensions/)? If yes, can you see if the workaround works for you?

**Comment by Kaelum** [comment]
@barrytang: Changing the setting doesn't change anything, it still throws the same exception.  We don't use Git, so it doesn't apply.

**Comment by Kaelum** [comment]
@mlorbetske, @barrytang, @madskristensen: After installing all of the latest VS updates, including the latest tool updates from Mads, with **Update 2 with ASP.NET 5 RC1 Update 1** (1.0.20204.0), the activity log has changed.  The errors are all occurring at the file preview stage, and here they are:

```
record:	1211
time:	2016/05/03 18:45:04.064
type:	Error
source:	Editor or Editor Extension

System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.VisualStudio.Web.Editors.Razor.DesignTimeHost.RazorDnxCommandInvoker.InvokeNoProtocolAdded(String cmdLine)
   at Microsoft.VisualStudio.Web.Editors.Razor.DesignTimeHost.RazorDnxCommandInvoker.Initialize(ITextBuffer textBuffer, IDnxCommandInvoker dnxCommandInvoker)
   at Microsoft.VisualStudio.Web.Editors.Razor.DesignTimeHost.RazorDnxCommandInvoker.Create(ITextBuffer textBuffer)
   at Microsoft.VisualStudio.Web.Editors.Razor.RazorCodeGenerator..ctor(ITextBuffer buffer, Version razorVersion, String physicalPath, String virtualPath)
   at Microsoft.VisualStudio.Web.Editors.Razor.RazorCodeGenerator.Create(ITextBuffer buffer, Version razorVersion, String physicalPath, String virtualPath)
   at Microsoft.VisualStudio.Web.Editors.Razor.RazorCodeGeneratorProvider.Microsoft.Html.Editor.Razor.IRazorCodeGeneratorProvider.CreateRazorCodeGenerator(ITextBuffer buffer, Version razorVersion, String physicalPath, String virtualPath)
   at Microsoft.Html.Editor.Razor.RazorCodeGeneratorFactory.CreateCodeGenerator(ITextBuffer buffer, Version razorVersion, String physicalPath, String virtualPath)
   at Microsoft.Html.Editor.ContentType.Handlers.RazorContentTypeHandler.Init(HtmlEditorTree editorTree)
   at Microsoft.Html.Editor.Document.HtmlEditorDocument..ctor(ITextBuffer textBuffer, IWebWorkspaceItem workspaceItem, Boolean disableContainedLanguages)
   at Microsoft.VisualStudio.Html.Package.Document.VsHtmlEditorDocument..ctor(IWebEditorInstance editorInstance, Boolean disableContainedLanguages)
   at Microsoft.VisualStudio.Html.Package.Commands.Html.VsHtmlTextViewConnectionListener.HtmlDocumentFactoryProjectionsDisabled.CreateDocument(IWebEditorInstance editorInstance)
   at Microsoft.Html.Editor.EditorFactory.HtmlEditorInstance..ctor(IWebWorkspaceItem workspaceItem, ITextBuffer diskBuffer, IWebEditorDocumentFactory documentFactory)
   at Microsoft.Html.Editor.EditorFactory.HtmlEditorFactory.CreateEditorInstance(IWebWorkspaceItem workspaceItem, Object textBuffer, IWebEditorDocumentFactory documentFactory)
   at Microsoft.VisualStudio.Html.Package.Commands.Html.VsHtmlTextViewConnectionListener.InitEditorInstance(ITextBuffer textBuffer)
   at Microsoft.VisualStudio.Html.Package.Commands.Html.VsHtmlTextViewConnectionListener.OnTextBufferCreated(ITextBuffer textBuffer)
   at Microsoft.Web.Editor.Controller.TextViewConnectionListener.SubjectBuffersConnected(IWpfTextView textView, ConnectionReason reason, Collection`1 subjectBuffers)
   at Microsoft.VisualStudio.Text.Editor.Implementation.ConnectionManager.&lt;&gt;c__DisplayClass4_1.&lt;.ctor&gt;b__1()
   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(Object errorSource, Action call)
```

```
record:	1212
time:	2016/05/03 18:45:08.410
type:	Error
source:	Editor or Editor Extension

System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.VisualStudio.Web.Editors.Razor.RazorCodeGenerator.Close()
   at Microsoft.VisualStudio.Web.Editors.Razor.RazorCodeGenerator.OnClose(Object sender, EventArgs args)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at Microsoft.Html.Editor.Document.HtmlEditorDocument.Close()
   at Microsoft.VisualStudio.Html.Package.Document.VsHtmlEditorDocument.Close()
   at Microsoft.Html.Editor.Document.HtmlEditorDocument.Dispose(Boolean disposing)
   at Microsoft.Html.Editor.Commands.HtmlTextViewConnectionListener.OnTextBufferDisposing(ITextBuffer textBuffer)
   at Microsoft.VisualStudio.Html.Package.Commands.Html.VsHtmlTextViewConnectionListener.OnTextBufferDisposing(ITextBuffer textBuffer)
   at Microsoft.Web.Editor.Controller.TextViewConnectionListener.CheckForViewlessTextBuffers()
   at Microsoft.Web.Editor.Controller.TextViewConnectionListener.SubjectBuffersDisconnected(IWpfTextView textView, ConnectionReason reason, Collection`1 subjectBuffers)
   at Microsoft.VisualStudio.Text.Editor.Implementation.ConnectionManager.&lt;&gt;c__DisplayClass5_1.&lt;Close&gt;b__1()
   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(Object errorSource, Action call)
```

```
record:	1213
time:	2016/05/03 18:45:08.432
type:	Error
source:	Editor or Editor Extension

System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.VisualStudio.Web.Editors.Razor.DesignTimeHost.RazorDnxCommandInvoker.InvokeNoProtocolAdded(String cmdLine)
   at Microsoft.VisualStudio.Web.Editors.Razor.DesignTimeHost.RazorDnxCommandInvoker.Initialize(ITextBuffer textBuffer, IDnxCommandInvoker dnxCommandInvoker)
   at Microsoft.VisualStudio.Web.Editors.Razor.DesignTimeHost.RazorDnxCommandInvoker.Create(ITextBuffer textBuffer)
   at Microsoft.VisualStudio.Web.Editors.Razor.RazorCodeGenerator..ctor(ITextBuffer buffer, Version razorVersion, String physicalPath, String virtualPath)
   at Microsoft.VisualStudio.Web.Editors.Razor.RazorCodeGenerator.Create(ITextBuffer buffer, Version razorVersion, String physicalPath, String virtualPath)
   at Microsoft.VisualStudio.Web.Editors.Razor.RazorCodeGeneratorProvider.Microsoft.Html.Editor.Razor.IRazorCodeGeneratorProvider.CreateRazorCodeGenerator(ITextBuffer buffer, Version razorVersion, String physicalPath, String virtualPath)
   at Microsoft.Html.Editor.Razor.RazorCodeGeneratorFactory.CreateCodeGenerator(ITextBuffer buffer, Version razorVersion, String physicalPath, String virtualPath)
   at Microsoft.Html.Editor.ContentType.Handlers.RazorContentTypeHandler.Init(HtmlEditorTree editorTree)
   at Microsoft.Html.Editor.Document.HtmlEditorDocument..ctor(ITextBuffer textBuffer, IWebWorkspaceItem workspaceItem, Boolean disableContainedLanguages)
   at Microsoft.VisualStudio.Html.Package.Document.VsHtmlEditorDocument..ctor(IWebEditorInstance editorInstance, Boolean disableContainedLanguages)
   at Microsoft.VisualStudio.Html.Package.Commands.Html.VsHtmlTextViewConnectionListener.HtmlDocumentFactoryProjectionsDisabled.CreateDocument(IWebEditorInstance editorInstance)
   at Microsoft.Html.Editor.EditorFactory.HtmlEditorInstance..ctor(IWebWorkspaceItem workspaceItem, ITextBuffer diskBuffer, IWebEditorDocumentFactory documentFactory)
   at Microsoft.Html.Editor.EditorFactory.HtmlEditorFactory.CreateEditorInstance(IWebWorkspaceItem workspaceItem, Object textBuffer, IWebEditorDocumentFactory documentFactory)
   at Microsoft.VisualStudio.Html.Package.Commands.Html.VsHtmlTextViewConnectionListener.InitEditorInstance(ITextBuffer textBuffer)
   at Microsoft.VisualStudio.Html.Package.Commands.Html.VsHtmlTextViewConnectionListener.OnTextBufferCreated(ITextBuffer textBuffer)
   at Microsoft.Web.Editor.Controller.TextViewConnectionListener.SubjectBuffersConnected(IWpfTextView textView, ConnectionReason reason, Collection`1 subjectBuffers)
   at Microsoft.VisualStudio.Text.Editor.Implementation.ConnectionManager.&lt;&gt;c__DisplayClass4_1.&lt;.ctor&gt;b__1()
   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(Object errorSource, Action call)
```

```
record:	1214
time:	2016/05/03 18:45:08.449
type:	Error
source:	Editor or Editor Extension

System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.VisualStudio.Web.Editors.Razor.RazorCodeGenerator.Close()
   at Microsoft.VisualStudio.Web.Editors.Razor.RazorCodeGenerator.OnClose(Object sender, EventArgs args)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at Microsoft.Html.Editor.Document.HtmlEditorDocument.Close()
   at Microsoft.VisualStudio.Html.Package.Document.VsHtmlEditorDocument.Close()
   at Microsoft.Html.Editor.Document.HtmlEditorDocument.Dispose(Boolean disposing)
   at Microsoft.Html.Editor.Commands.HtmlTextViewConnectionListener.OnTextBufferDisposing(ITextBuffer textBuffer)
   at Microsoft.VisualStudio.Html.Package.Commands.Html.VsHtmlTextViewConnectionListener.OnTextBufferDisposing(ITextBuffer textBuffer)
   at Microsoft.Web.Editor.Controller.TextViewConnectionListener.CheckForViewlessTextBuffers()
   at Microsoft.Web.Editor.Controller.TextViewConnectionListener.SubjectBuffersDisconnected(IWpfTextView textView, ConnectionReason reason, Collection`1 subjectBuffers)
   at Microsoft.VisualStudio.Text.Editor.Implementation.ConnectionManager.&lt;&gt;c__DisplayClass5_1.&lt;Close&gt;b__1()
   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(Object errorSource, Action call)
```

**Comment by Kaelum** [comment]
@mlorbetske, @barrytang, @madskristensen: EUREKA!

Looking back at @tobiasbunyan 's post, I looked at our Web.config which **HAD** (the EUREKA purpose) the following:

```
	<appSettings>
		<add key="webpages:Version" value="4.0.0.0" />
		<add key="webpages:Enabled" value="false" />
		<!-- other settings -->
	</appSettings>
```

Apparently, Visual Studio does not like those 2 entries any longer, even though **Enabled** is set to **false**.  If I change the version to **3.0.0.0**, or delete the entries altogether, IT WORKS!  I know that I tried this when @tobiasbunyan first posted this, but the build at that time still wasn't working.  With the latest updates, it is now working, and removing the lines doesn't appear to be breaking Visual Studio 2013, which is where the project was originally created.

**Comment by tbunyan-winman** [comment]
IIRC taking the entries out altogether will just make VS try and use the 'nearest' version that it can.  4.0.0.0 is apparently the version distributed with .NET 5 RC1, 3.x being that for for .NET 4.  I wouldn't have thought the enabled flag would make any difference anyway as VS will still need the engine to parse the files when you open them.

The razor version. however, I believe should be independent of the version of MVC you are using i.e. you can use Razor 3.x with .NET 5 RC1.  Someone else might be able to clarify this better though.

So you haven't strictly solved the problem I don't think, but at least you can edit the files now and get some work done ;)

Why mine had 5.0 in is anyones guess....

**Comment by tobiasbunyan** [comment]
IIRC taking the entries out altogether will just make VS try and use the 'nearest' version that it can.  4.0.0.0 is apparently the version distributed with ASP .NET 5 RC1, 3.x being that for for ASP .NET 4.  I wouldn't have thought the enabled flag would make any difference anyway as VS will still need the engine to parse the files when you open them.

The razor version, however, I believe should be independent of the version of MVC you are using i.e. you can use Razor 3.x with ASP .NET 5 RC1.  Someone else might be able to clarify this better though.

So you haven't strictly solved the problem I don't think, but at least you can edit the files now and get some work done ;)

Why mine had 5.0 in is anyones guess....

**Comment by Kaelum** [comment]
Hey @tobiasbunyan, this has been a sore point in MVC since the settings were created.  The "webpages:Version" appSetting should never have been an appSetting in the first place, and in fact, this setting was rendered unneeded prior to VS 2013's RTM.  Instead, the .csproj tooling should have been updated to support the setting, if created at all.

This is truly a "Design time" setting even though I understand the thoughts, at the time, for using this as a runtime setting as well, for backwards compatibility.  The amount of time required to upgrade MVC 3 or 4 views to MVC 5 views was so insignificant that it truly rendered this setting useless.

As for how it got changed to "4.0.0.0", TFS shows that it happened while upgrading several NuGet updates about a year ago, so one of them may have changed it when it shouldn't have.  At least we have a work-around now, as people were becoming extremely concerned that we couldn't do side-by-side development in VS 2015.

**Comment by sayedihashimi** [comment]
TFS Bug#<a href="https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workItems#id=231169&fullScreen=false&_a=edit">231169</a>

**Comment by Gaby10** [comment]
Still experimenting same problems (can't edit a cshtml file) after starting an MVC project. I'm using Windows enterprise 10, Build 14372, I've installed all the updates and post Update 3 patch/kb... it was working days ago. I'll try all the solutions mentioned here, my gut tells me that is as version problem in config files... I'll start there. But I wanted to let you know that this problem persists even today.

**Comment by hbrock** [comment]
This makes me want to move to Linux 100%. Microsoft seem to not care about its customer anymore.

**Comment by markau** [comment]
So hows this issue going?

**Comment by hbrock** [comment]
@markau I had to re-install windows 10 to fix the problem.

**Comment by Kaelum** [comment]
The issue that I reported has a work-around, and I believe that Mads has already updated the tooling to correct the issue.  If you have an issue that isn't fixed by correcting bad web.config values, I would suggest opening a new ticket, as this ticket is for a different issue.

**Comment by markau** [comment]
I think the same issue has different causes, going back to at least September 2015 from what I've seen on StackOverflow. 

Pretty sure the ASP.NET 5 tooling is the cause of mine. VS2015 is still broken, and I don't see a commit linked to this issue, so I thought it was reasonable to vent my frustration here.

**Comment by markau** [comment]
My apologies; the tooling has indeed been updated to fix this issue with release 2. I was unable to install that due to an issue with SP3, however, until I saw [this SO question and answer](https://stackoverflow.com/questions/38684594/setup-has-detected-that-visual-studio-2015-update-3-may-not-be-completely-instal). 

I'm glad to be able to get back to work. Thanks.

**Comment by tdriver** [comment]
This issue just popped up with me, for the first time.  I has started to play with dotnet core and installed DotNetCore.1.0.1-VS2015Tools.Preview2.0.3.exe on my VS 2015 Update 3 baseline.  It looks like (from the Uninstall panel) that ASP.Net 5 RC1 Update 1 got updated at the same time, based on the "Installed On" date.  After I installed, cshtml files would not open - giving me an MessageBox error that said the file has been removed or deleted.  I scoured these issues and tried all the recommended fixes - no joy.  I finally looked at my Extensions and Updates in VS and noticed that it wanted me to install the latest Azure SDK update. I did that, and now I can open cshtml files again.  I checked my web.configs on the projects that were affected and I do have webpages:Version = 3.0.0.0.  I didn't get around to modifying those to see if that fixed it.
Just posting to let you know that this problem is still out there, randomly popping up when tooling updates appear to be out of sync.

**Comment by ygoe** [comment]
> DotNetCore.1.0.1-VS2015Tools.Preview2.0.3.exe

> ASP.Net 5 RC1 Update 1

This is insane. When will Microsoft learn what [version numbers](http://semver.org/) are?

**Comment by Borzakov** [comment]
Installing "tools" helped 
No proper name of tools (VS2015 Preview 2 tooling for Microsoft .NET Core 1.0.1) or [link](https://marketplace.visualstudio.com/items?itemName=JacquesEloff.MicrosoftASPNETandWebTools-9689) to them were given in this discussion, so here they are.

**Comment by dipakdipak** [comment]
​I found the fix to this problem by disabling the PHP Tools extension (and closing/restarting Visual studio). It is working for me now. May be other extension can create this issue for others.
